punx/tests/test_nxdl_manager.py

```diff
@@ -7,29 +7,78 @@
 from .. import FileNotFound
 from .. import InvalidNxdlFile
 from .. import nxdl_manager
+from .. import nxdl_schema
 
 
-def raises_exception(text):
-    try:
-        lxml.etree.fromstring(text)
-    except lxml.etree.XMLSyntaxError as exc:
-        raise exc
-    except Exception as exc:
-        raise exc
-    else:
-        raise No_Exception
+def get_NXDL_definition(nxclass, file_set):
+    """
+    Return instance of nxdl_manager.NXDL__definition from file_set.
+
+    PARAMETERS
+
+    nxclass str:
+        Name of NeXus class (such as NXdata, NXentry, NXsas, ...)
+    file_set str:
+        Name of NeXus definitions *file set* to be found in
+        one of the local caches.  A file set is a directory
+        containing all the NXDL files and XML Schema files
+        that define a specific version of the NeXus definitions.
+    """
+    cache_manager.CacheManager()
+    manager = nxdl_manager.NXDL_Manager(file_set)
+    return manager.classes.get(nxclass)
+
+
+def navigate_path(path_start, nxpath):
+    """
+    Drill down the nxpath to the group containing the last item.
+
+    PARAMETERS
+
+    path_start obj:
+        Instance of either ``nxdl_manager.NXDL__definition``
+        or ``nxdl_manager.NXDL__group`` that serves as the starting
+        point for ``nxpath``.
+    nxpath str:
+        NeXus path specification to an entity defined in a NXDL file.
+        This is not necessarily an HDF5 address in an HDF5 data file
+        but it sure looks like one.  Pay particular attention to the
+        use of default names for unnamed groups.  Unnamed groups are
+        common in NXDL files.
+
+        Example: ``/entry/data/data``
+    """
+    group = path_start  # starting point
+
+    parts = nxpath.lstrip("/").split("/")
```

Comment:
Consider using `os.path` module? I'm not sure that `/` is compatible with all platform paths.

Reply:
Good eye here.  Adding a comment to explain why os-specific handling should not be used.